packages/react-sdk/src/prop-transforms/image.ts

```diff
@@ -0,0 +1,87 @@
+import type { TransformProps, UserProp } from "~/user-props";
+import { getImageAttributes } from "./image-optimizer";
+import {
+  cloudflareImageLoader,
+  localImageLoader,
+  type CloudflareImageLoaderOptions,
+} from "../prop-transforms/image-loaders";
+
+const ensureString = (value: unknown): string | undefined => {
+  if (typeof value === "string") return value;
+  return undefined;
+};
+
+const ensureStringOrNumber = (value: unknown): number | string | undefined => {
+  if (typeof value === "string") return value;
+  if (typeof value === "number") return value;
+  return undefined;
+};
+
+const ensureBoolean = (value: unknown): boolean | undefined => {
+  if (typeof value === "boolean") return value;
+  return undefined;
+};
+
+let gId = 0;
+
+export type ImageTransformOptions = CloudflareImageLoaderOptions;
+
+export const createPropsTransformer: (
+  opts: CloudflareImageLoaderOptions | null
+) => TransformProps = (options) => (props) => {
+  const res: UserProp[] = [];
+
+  const imageValueProps: Record<string, unknown> = {};
+
+  for (const prop of props) {
+    if ("value" in prop) {
+      res.push(prop);
+      imageValueProps[prop.prop] = prop.value;
+    }
+  }
+
+  for (const prop of props) {
+    if ("asset" in prop) {
+      if (prop.prop !== "src") {
+        throw new Error("Image component can only accept src prop as asset");
+      }
+
+      const { asset } = prop;
+
+      let loader = localImageLoader;
+
+      // import { Location } from "@webstudio-is/prisma-client" cause remix to fail
+      // with "Cannot initialize 'routeModules'" error
+      // https://remix.run/docs/en/v1/pages/gotchas#server-code-in-client-bundles
+      // use "REMOTE" const instead of Location.REMOTE
+      if (asset.location === "REMOTE") {
+        loader = cloudflareImageLoader(options);
+      } else {
+        loader = localImageLoader;
+      }
+
+      const imageAttributes = getImageAttributes({
+        src: asset.path,
+        srcSet: ensureString(imageValueProps.srcSet),
+        sizes: ensureString(imageValueProps.sizes),
+        width: ensureStringOrNumber(imageValueProps.width),
+        quality: ensureStringOrNumber(imageValueProps.quality),
+        loader,
+        // User can overwrite optimize
+        optimize: ensureBoolean(imageValueProps.optimize) ?? true,
+      });
+
+      if (imageAttributes != null) {
+        for (const [key, value] of Object.entries(imageAttributes)) {
+          res.push({
+            id: `imgt_${gId++}`,
```

Comment:
What `imgt` means and what `g` means in `gId`? 😅

Reply:
Just need some id non intersected with others. 
g means global ;-) If we have some convention about mod level state vars tell me. 
Usualy I use underscore after var, i.e. `id_`

Reply:
Can we do something like `image_transform_${globalId}` so it would be less confusing? (if I guessed correctly what `imgt` means)

Reply:
Done